src/analytics/Asfans.js

```diff
@@ -0,0 +1,207 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+
+import { Col, Row, Table, InputGroup, InputGroupAddon, InputGroupText, CustomInput, NavLink } from 'reactstrap';
+
+import { getDraftFormat, matchingCards } from 'utils/draftutil';
+import { sortIntoGroups, getSorts } from 'utils/Sort';
+import ErrorBoundary from 'components/ErrorBoundary';
+
+const useSortableData = (items, config = null) => {
+  const [sortConfig, setSortConfig] = useState(config);
+
+  const sortedItems = React.useMemo(() => {
+    const sortableItems = [...items];
+    if (sortConfig !== null) {
+      sortableItems.sort((a, b) => {
+        if (a[sortConfig.key] - b[sortConfig.key] < 0) {
+          return sortConfig.direction === 'ascending' ? -1 : 1;
+        }
+        if (a[sortConfig.key] - b[sortConfig.key] > 0) {
+          return sortConfig.direction === 'ascending' ? 1 : -1;
+        }
+        return 0;
+      });
+    }
+    return sortableItems;
+  }, [items, sortConfig]);
+
+  const requestSort = (key) => {
+    let direction = 'descending';
+    if (sortConfig && sortConfig.key === key && sortConfig.direction === 'descending') {
+      direction = 'ascending';
+    }
+    setSortConfig({ key, direction });
+  };
+
+  return { items: sortedItems, requestSort, sortConfig };
+};
+
+const HeaderCell = ({ label, fieldName, sortConfig, requestSort }) => {
+  let icon = '/content/nosort.png';
+
+  if (sortConfig && sortConfig.key === fieldName) {
+    if (sortConfig.direction === 'descending') {
+      icon = '/content/ascending.png';
+    } else {
+      icon = '/content/descending.png';
+    }
+  }
+
+  return (
+    <th scope="col">
+      <NavLink href="#" onClick={() => requestSort(fieldName)}>
+        {label} <img src={icon} className="sortIcon" alt="" />
+      </NavLink>
+    </th>
+  );
+};
+
+HeaderCell.propTypes = {
+  label: PropTypes.string.isRequired,
+  fieldName: PropTypes.string.isRequired,
+  sortConfig: PropTypes.shape({
+    key: PropTypes.string.isRequired,
+    direction: PropTypes.string.isRequired,
+  }).isRequired,
+  requestSort: PropTypes.string.isRequired,
+};
+
+const Asfans = ({ cards, cube }) => {
+  const sorts = getSorts();
+
+  const [sort, setSort] = useState('Color');
+  const [formatId, setFormatId] = useState(-1);
+
+  const draftFormats = cube.draft_formats;
+
+  const calculateAsfans = () => {
+    return [
+      {
+        label: '',
+        data: Object.entries(sortIntoGroups(cards, sort)).map((tuple) => {
+          return {
+            label: tuple[0],
+            asfan: (tuple[1].length / cube.cards.length) * 15, // 15 cards a pack
+          };
+        }),
+      },
+    ];
+  };
+
+  const calculateCustomAsfans = (draftFormat) => {
```

Comment:
I don't believe this will properly handle joins where a card falls into multiple different filters. Such as "rarity:Common" and "type:Land" in the case you have common lands in the cube. In that case the chance you pick it is `1/len(Commons) + (1 - 1/len(Commons))/len(Lands)` since it needs to have not been picked yet.

Reply:
That's fair, I think the margin of error is incredibly small though, do you want to make a follow up PR that updates the Math here? I only reworked it because the old system made 0 sense to me.

Reply:
I'll work on it in a followup.

Reply:
Actually the solution was super easy. Instead of doing a sum with total, I did this:
` total += (1-total) * sum / slot.length;`

Reply:
That's closer but still fails the case where another card is in multiple since length depends on if that was taken already.